pgmpy/continuous/discretize/base.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+from pgmpy.factors import ContinuousFactor
+
+
+class BaseDiscretizer(object):
+    """
+    Base class for the discretizer classes in pgmpy. The discretizer
+    classes are used to discretize a continuous random variable
+    distribution into discrete probability masses.
+
+    Parameters
+    ----------
+    factor: ContinuousFactor
+        the continuous factor representing the distribution
+        to be discretized.
+    frm, to: float
+        the range over which the function will be discretized.
+    step: float
+        the discretization step (or span, or lag)
+
+    """
+
+    def __init__(self, factor, frm, to, step):
```

Comment:
@yashu-seth I don't think `step` is a required parameter for every discretization function. Think of some method that samples only from high probability regions, the step size will vary for each sample. Having a parameter for no of samples (which would be the states) makes more sense to me. What do you think ?